### PR TITLE
Fix/dot format bug

### PR DESCRIPTION
**Pull request by atravitz**
Fixes issue #375. 

## Description
- Rename `ProjectSyncTest` to `TestProjectSync` so pytest recognizes the tests.
- Correct a typo in checking difference between `src` and `dst` in the CLI sync
- Correct the formatting for CLI sync error output.

## Motivation and Context
see #375 

## Types of Changes

- [ ] Documentation update
- [x] Bug fix
- [ ] New feature
- [ ] Breaking change<sup>1</sup>

<sup>1</sup>The change breaks (or has the potential to break) existing functionality.

## Checklist:

- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac/blob/master/contributors.yaml).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac/blob/master/CONTRIBUTING.md#code-style) of this project.
- [ ] The changes introduced by this pull request are covered by existing or newly introduced tests.

If necessary:
- [ ] I have updated the API documentation as part of the package doc-strings.
- [ ] I have created a separate pull request to update the [framework documentation](https://docs.signac.io/) on [signac-docs](https://github.com/glotzerlab/signac-docs) and linked it here.
- [ ] I have updated the [changelog](https://github.com/glotzerlab/signac/blob/master/changelog.txt) and added all related issue and pull request numbers for future reference (if applicable). See example below.

Example for a changelog entry: `Fix issue with launching rockets to the moon (#101, #212).`

### CODE REVIEW

File: signac/__main__.py [hunk 1]
```diff
@@ -568,20 +568,17 @@ def _sig(st):
         diff_src = error.schema_src.difference(error.schema_dst)
         diff_dst = error.schema_dst.difference(error.schema_src)
         only_in_dst = diff_dst.difference(diff_src)
-        only_in_src = diff_src.difference(diff_src)
+        only_in_src = diff_src.difference(diff_dst)
         diff_value = diff_src.intersection(diff_dst)
         if only_in_src:
-            keys_formatted = ('.'.join(k) for k in only_in_src)
```
Comment (vyasr) [inline signac/__main__.py:574]: @atravitz why did you remove this? Are nested statepoints being converted from a list representation to a dot-separated string upstream of this?

Comment (bdice) [reply signac/__main__.py:574]: I confirmed that the keys are converted to a dot-separated string. This is safe to remove.
